docs/src/demos/rgb_hsv_thresholding.md

```diff
@@ -0,0 +1,37 @@
+# [RGB to HSV and thresholding](@id rgb_hsv_thresholding)
+
+This example illustrates how RGB to HSV (Hue, Saturation, Value) conversion
+can be used to facilitate segmentation processes.
+A simple segmentation of the image can then be effectively performed 
+by a mere thresholding of the HSV channels.
+
+
+```julia
+using Images, TestImages, LinearAlgebra, Interact, ImageView
+
+rbg_img = testimage("lighthouse")
+hsv_img = HSV.(rbg_img)
+channels = float(channelview(img))
+
+hue_img = channels[1,:,:]
+value_img = channels[3,:,:]
+saturation_img = channels[2,:,:]
+@manipulate for hue in 0:255, saturation in 0:255, value in 0:255
+    binary_img = zeros(size(hue_img))
```

Comment:
can this be preallocated outside of `@manipulate` or does that break the code somehow?

Reply:
@Evizero It won't work outside, the `binary_img` will change after each thresholding and the array needs to be reset each time.

Reply:
right. I was more talking about moving the allocation outside and using `fill!(binary_img, 0.0)` inside instead (which doesn't allocate)

Reply:
@Evizero I wasn't quite aware the method. I pushed the suggested changes. 
Just a question out of curiosity,
Does this speeds up the process or is it according to coding standards?

Reply:
well `zeros()` allocates a new array every time it is called and then fills that array with zeros.
this new approach reuses the same array every time and just resets the values. In pretty much all cases that i encountered this kind of preallocation will be faster